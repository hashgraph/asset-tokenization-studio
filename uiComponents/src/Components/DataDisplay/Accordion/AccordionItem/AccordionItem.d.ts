import type { AccordionItemProps as ChakraAccordionItemProps } from "@chakra-ui/react";
import type { SystemStyleObject as ChakraSystemStyleObject, ComponentWithAs } from "@chakra-ui/system";
import React from "react";
import type { BaseMultiStyleConfiguration, PartsStyleInterpolation } from "@/Theme/Components/BaseMultiStyleConfiguration";
export interface AccordionItemProps extends Omit<ChakraAccordionItemProps, "children" | "title"> {
    children: React.ReactNode;
    title?: React.ReactNode;
    icon?: any;
    customTitle?: React.ReactNode;
}
declare const accordionItemPartsList: Array<string>;
type Parts = typeof accordionItemPartsList;
export interface AccordionItemThemeConfiguration extends BaseMultiStyleConfiguration<Parts> {
    baseStyle: (() => Partial<AccordionItemThemeStyle>) | PartsStyleInterpolation<Parts>;
}
type AccordionItemThemeStyle = Record<Parts[number], ChakraSystemStyleObject>;
export declare const AccordionItem: ComponentWithAs<"div", AccordionItemProps>;
export {};
