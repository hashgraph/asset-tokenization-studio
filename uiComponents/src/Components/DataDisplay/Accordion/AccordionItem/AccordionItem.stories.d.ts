import React from "react";
declare const meta: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, any, import("./AccordionItem").AccordionItemProps, import("@chakra-ui/system/dist/system.types").As>>;
export default meta;
export declare const DefaultIcon: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, any, import("./AccordionItem").AccordionItemProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const CustomIcon: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, any, import("./AccordionItem").AccordionItemProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const CustomTitleComponent: () => JSX.Element;
export declare const CustomTitle: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, any, import("./AccordionItem").AccordionItemProps, import("@chakra-ui/system/dist/system.types").As>>;
