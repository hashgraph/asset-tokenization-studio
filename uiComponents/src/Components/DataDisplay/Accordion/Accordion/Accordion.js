var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { Accordion as ChakraAccordion } from "@chakra-ui/accordion";
import { Box as ChakraBox, Divider as ChakraDivider } from "@chakra-ui/layout";
import { forwardRef, useMultiStyleConfig as useChakraMultiStyleConfig, } from "@chakra-ui/system";
import React from "react";
import { Text } from "@/Components/Foundations/Text";
var accordionPartsList = ["container", "title", "item"];
export var Accordion = forwardRef(function (_a, ref) {
    var children = _a.children, description = _a.description, title = _a.title, props = __rest(_a, ["children", "description", "title"]);
    var styles = useChakraMultiStyleConfig("Accordion");
    return (React.createElement(ChakraAccordion, __assign({ ref: ref }, props, { sx: styles.container }),
        React.createElement(ChakraBox, { sx: styles.titleContainer },
            React.createElement(Text, { sx: styles.title }, title)),
        React.createElement(ChakraDivider, { sx: styles.divider }),
        description && (React.createElement(ChakraBox, { sx: styles.descriptionContainer },
            React.createElement(Text, { sx: styles.description }, description))),
        React.createElement(ChakraBox, { sx: styles.itemsContainer }, children)));
});
