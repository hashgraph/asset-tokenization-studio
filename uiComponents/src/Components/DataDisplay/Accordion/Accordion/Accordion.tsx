import type { AccordionProps as ChakraAccordionProps } from "@chakra-ui/react";
import { Accordion as ChakraAccordion } from "@chakra-ui/accordion";
import { Box as ChakraBox, Divider as ChakraDivider } from "@chakra-ui/layout";
import type {
  SystemStyleObject as ChakraSystemStyleObject,
  ComponentWithAs,
} from "@chakra-ui/system";
import {
  forwardRef,
  useMultiStyleConfig as useChakraMultiStyleConfig,
} from "@chakra-ui/system";
import React from "react";
import type {
  BaseMultiStyleConfiguration,
  PartsStyleInterpolation,
} from "@/Theme/Components/BaseMultiStyleConfiguration";
import { Text } from "@/Components/Foundations/Text";

export interface AccordionProps
  extends Omit<ChakraAccordionProps, "children" | "title" | "description"> {
  children: React.ReactNode;
  description?: string;
  title: string;
}

const accordionPartsList: Array<string> = ["container", "title", "item"];

type Parts = typeof accordionPartsList;

export interface AccordionThemeConfiguration
  extends BaseMultiStyleConfiguration<Parts> {
  baseStyle:
    | (() => Partial<AccordionThemeStyle>)
    | PartsStyleInterpolation<Parts>;
}

export type AccordionThemeStyle = Record<
  Parts[number],
  ChakraSystemStyleObject
>;

export const Accordion: ComponentWithAs<"div", AccordionProps> = forwardRef<
  AccordionProps,
  "div"
>(({ children, description, title, ...props }: AccordionProps, ref) => {
  const styles = useChakraMultiStyleConfig("Accordion");

  return (
    <ChakraAccordion ref={ref} {...props} sx={styles.container}>
      <ChakraBox sx={styles.titleContainer}>
        <Text sx={styles.title}>{title}</Text>
      </ChakraBox>
      <ChakraDivider sx={styles.divider} />
      {description && (
        <ChakraBox sx={styles.descriptionContainer}>
          <Text sx={styles.description}>{description}</Text>
        </ChakraBox>
      )}
      <ChakraBox sx={styles.itemsContainer}>{children}</ChakraBox>
    </ChakraAccordion>
  );
});
