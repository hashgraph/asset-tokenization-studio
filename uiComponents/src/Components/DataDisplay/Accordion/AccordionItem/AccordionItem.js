var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { AccordionIcon as ChakraAccordionIcon, AccordionItem as ChakraAccordionItem, AccordionButton as ChakraAccordionButton, AccordionPanel as ChakraAccordionPanel, } from "@chakra-ui/react";
import { Box as ChakraBox } from "@chakra-ui/layout";
import { forwardRef, useMultiStyleConfig as useChakraMultiStyleConfig, } from "@chakra-ui/system";
import React from "react";
var accordionItemPartsList = ["button", "item", "panel"];
export var AccordionItem = forwardRef(function (_a, ref) {
    var children = _a.children, customTitle = _a.customTitle, title = _a.title, icon = _a.icon, props = __rest(_a, ["children", "customTitle", "title", "icon"]);
    var styles = useChakraMultiStyleConfig("AccordionItem");
    return (React.createElement(ChakraAccordionItem, __assign({ ref: ref }, props, { sx: styles.container }),
        title && (React.createElement("h2", null,
            React.createElement(ChakraAccordionButton, { sx: styles.button },
                React.createElement(ChakraBox, { sx: styles.title }, title),
                React.createElement(ChakraAccordionIcon, __assign({}, (icon && { as: icon })))))),
        customTitle && React.createElement(ChakraBox, { w: "full" }, customTitle),
        React.createElement(ChakraAccordionPanel, { sx: styles.panel }, children)));
});
