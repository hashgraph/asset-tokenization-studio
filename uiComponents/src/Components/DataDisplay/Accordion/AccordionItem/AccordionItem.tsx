import type { AccordionItemProps as ChakraAccordionItemProps } from "@chakra-ui/react";
import {
  AccordionIcon as ChakraAccordionIcon,
  AccordionItem as ChakraAccordionItem,
  AccordionButton as ChakraAccordionButton,
  AccordionPanel as ChakraAccordionPanel,
} from "@chakra-ui/react";
import { Box as ChakraBox } from "@chakra-ui/layout";
import type {
  SystemStyleObject as ChakraSystemStyleObject,
  ComponentWithAs,
} from "@chakra-ui/system";
import {
  forwardRef,
  useMultiStyleConfig as useChakraMultiStyleConfig,
} from "@chakra-ui/system";
import React from "react";
import type {
  BaseMultiStyleConfiguration,
  PartsStyleInterpolation,
} from "@/Theme/Components/BaseMultiStyleConfiguration";

export interface AccordionItemProps
  extends Omit<ChakraAccordionItemProps, "children" | "title"> {
  children: React.ReactNode;
  title?: React.ReactNode;
  icon?: any;
  customTitle?: React.ReactNode;
}

const accordionItemPartsList: Array<string> = ["button", "item", "panel"];

type Parts = typeof accordionItemPartsList;

export interface AccordionItemThemeConfiguration
  extends BaseMultiStyleConfiguration<Parts> {
  baseStyle:
    | (() => Partial<AccordionItemThemeStyle>)
    | PartsStyleInterpolation<Parts>;
}

type AccordionItemThemeStyle = Record<Parts[number], ChakraSystemStyleObject>;

export const AccordionItem: ComponentWithAs<"div", AccordionItemProps> =
  forwardRef<AccordionItemProps, "div">(
    (
      { children, customTitle, title, icon, ...props }: AccordionItemProps,
      ref
    ) => {
      const styles = useChakraMultiStyleConfig("AccordionItem");

      return (
        <ChakraAccordionItem ref={ref} {...props} sx={styles.container}>
          {title && (
            <h2>
              <ChakraAccordionButton sx={styles.button}>
                <ChakraBox sx={styles.title}>{title}</ChakraBox>
                <ChakraAccordionIcon {...(icon && { as: icon })} />
              </ChakraAccordionButton>
            </h2>
          )}
          {customTitle && <ChakraBox w="full">{customTitle}</ChakraBox>}
          <ChakraAccordionPanel sx={styles.panel}>
            {children}
          </ChakraAccordionPanel>
        </ChakraAccordionItem>
      );
    }
  );
