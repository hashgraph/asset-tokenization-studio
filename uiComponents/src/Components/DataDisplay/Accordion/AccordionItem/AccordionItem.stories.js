var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { AccordionItem } from ".";
import { Accordion } from "../Accordion";
import { ArrowDown } from "@phosphor-icons/react";
import { Tag } from "../../Tag";
import { HStack as ChakraHStack, AccordionButton as ChakraAccordionButton, AccordionIcon as ChakraAccordionIcon, Button as ChakraButton, } from "@chakra-ui/react";
var meta = {
    title: "Design System/Data Display/Accordion/AccordionItem",
    component: AccordionItem,
    argTypes: {
        variant: {}
    },
    parameters: {
        design: {},
        docs: {}
    },
    args: {
        children: React.createElement("p", null, "This is the item content")
    }
};
export default meta;
var Template = function (args) {
    return (React.createElement(Accordion, { title: "Accordion Title" },
        React.createElement(AccordionItem, __assign({}, args))));
};
export var DefaultIcon = Template.bind({});
DefaultIcon.args = {
    title: "Accordion Item Title"
};
export var CustomIcon = Template.bind({});
CustomIcon.args = {
    title: "Accordion Item with custom icon from Phosphor",
    icon: ArrowDown
};
export var CustomTitleComponent = function () { return (React.createElement(ChakraHStack, { w: "full", justify: "space-between", px: 4, "data-testid": "custom-title" },
    React.createElement(ChakraHStack, null,
        React.createElement(ChakraAccordionButton, { "data-testid": "custom-title-button" },
            React.createElement("h2", null, "Admin"),
            React.createElement(ChakraAccordionIcon, null)),
        React.createElement(ChakraHStack, { w: "auto" },
            React.createElement(Tag, { label: "Complement", size: "sm" }))),
    React.createElement(ChakraHStack, null,
        React.createElement(ChakraButton, { size: "sm", onClick: function () { return alert("You click on me"); } }, "Click")))); };
export var CustomTitle = Template.bind({});
CustomTitle.args = {
    customTitle: React.createElement(CustomTitleComponent, null)
};
