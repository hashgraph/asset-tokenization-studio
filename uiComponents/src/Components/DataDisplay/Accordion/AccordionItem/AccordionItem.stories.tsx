import type { Meta, StoryFn } from "@storybook/react";
import React from "react";
import { AccordionItem } from ".";
import { Accordion } from "../Accordion";
import { ArrowDown } from "@phosphor-icons/react";
import { Tag } from "../../Tag";
import {
  HStack as ChakraHStack,
  AccordionButton as ChakraAccordionButton,
  AccordionIcon as ChakraAccordionIcon,
  Button as ChakraButton,
} from "@chakra-ui/react";

const meta = {
  title: "Design System/Data Display/Accordion/AccordionItem",
  component: AccordionItem,
  argTypes: {
    variant: {},
  },
  parameters: {
    design: {},
    docs: {},
  },
  args: {
    children: <p>This is the item content</p>,
  },
} as Meta<typeof AccordionItem>;
export default meta;

const Template: StoryFn<typeof AccordionItem> = (args) => {
  return (
    <Accordion title="Accordion Title">
      <AccordionItem {...args} />
    </Accordion>
  );
};

export const DefaultIcon = Template.bind({});
DefaultIcon.args = {
  title: "Accordion Item Title",
};

export const CustomIcon = Template.bind({});
CustomIcon.args = {
  title: "Accordion Item with custom icon from Phosphor",
  icon: ArrowDown,
};

export const CustomTitleComponent = () => (
  <ChakraHStack
    w="full"
    justify="space-between"
    px={4}
    data-testid="custom-title"
  >
    <ChakraHStack>
      <ChakraAccordionButton data-testid="custom-title-button">
        <h2>Admin</h2>
        <ChakraAccordionIcon />
      </ChakraAccordionButton>
      <ChakraHStack w="auto">
        <Tag label="Complement" size="sm" />
      </ChakraHStack>
    </ChakraHStack>
    <ChakraHStack>
      <ChakraButton size="sm" onClick={() => alert("You click on me")}>
        Click
      </ChakraButton>
    </ChakraHStack>
  </ChakraHStack>
);

export const CustomTitle = Template.bind({});
CustomTitle.args = {
  customTitle: <CustomTitleComponent />,
};
