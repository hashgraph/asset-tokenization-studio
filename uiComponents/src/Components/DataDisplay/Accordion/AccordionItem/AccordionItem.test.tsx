import { render } from "@/test-utils";
import { Text } from "@chakra-ui/react";
import { waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React from "react";
import type { AccordionItemProps } from "./AccordionItem";
import { AccordionItem } from "./AccordionItem";
import { Accordion } from "../Accordion";
import { CustomTitleComponent } from "./AccordionItem.stories";

const text = "Example children";
const title = "Accordion Item Title";

const defaultProps: AccordionItemProps = {
  title,
  children: <Text>{text}</Text>,
};

describe(`<AccordionItem />`, () => {
  const factoryComponent = (props: AccordionItemProps = defaultProps) =>
    render(
      <Accordion title="Accordion Test">
        <AccordionItem {...props} />
      </Accordion>
    );

  test("should render correctly", () => {
    const component = factoryComponent();

    expect(component.getByText(title)).toBeVisible();
    expect(component).toMatchSnapshot();
  });

  test("onclick should expand and show children", async () => {
    const component = factoryComponent();

    expect(component.queryByText(text)).not.toBeVisible();

    userEvent.click(component.getByText(title));

    await waitFor(
      () => {
        expect(component.getByText(text)).toBeVisible();
      },
      { timeout: 50_000 }
    );
  });

  test("should render custom title", async () => {
    const customProps = {
      children: <Text>{text}</Text>,
      customTitle: <CustomTitleComponent />,
    };
    const component = factoryComponent(customProps);

    expect(component.getByTestId("custom-title")).toBeInTheDocument();
    expect(component.asFragment()).toMatchSnapshot("customTitle");
  });
});
