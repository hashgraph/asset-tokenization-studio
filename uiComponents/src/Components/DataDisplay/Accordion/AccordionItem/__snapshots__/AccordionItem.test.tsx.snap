// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionItem /> should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div>
      <div
        class="chakra-accordion css-1wbdjbz"
      >
        <div
          class="css-0"
        >
          <p
            class="css-0"
          >
            Accordion Test
          </p>
        </div>
        <hr
          aria-orientation="horizontal"
          class="chakra-divider css-svjswr"
        />
        <div
          class="css-0"
        >
          <div
            class="chakra-accordion__item css-17mg6aq"
          >
            <h2>
              <button
                aria-controls="accordion-panel-:r0:"
                aria-expanded="false"
                class="chakra-accordion__button css-uttm9k"
                data-index="0"
                id="accordion-button-:r0:"
                type="button"
              >
                <div
                  class="css-0"
                >
                  Accordion Item Title
                </div>
                <svg
                  aria-hidden="true"
                  class="chakra-icon chakra-accordion__icon css-186l2rg"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </h2>
            <div
              class="chakra-collapse"
              style="overflow: hidden; display: none; opacity: 0; height: 0px;"
            >
              <div
                aria-labelledby="accordion-button-:r0:"
                class="chakra-accordion__panel css-mxa4qn"
                id="accordion-panel-:r0:"
                role="region"
              >
                <p
                  class="chakra-text css-0"
                >
                  Example children
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <span
        hidden=""
        id="__chakra_env"
      />
    </div>
    <div
      class="chakra-portal"
    >
      <div
        aria-live="polite"
        id="chakra-toast-manager-top"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <div
        aria-live="polite"
        id="chakra-toast-manager-top-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-live="polite"
        id="chakra-toast-manager-top-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-live="polite"
        id="chakra-toast-manager-bottom-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-live="polite"
        id="chakra-toast-manager-bottom"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <div
        aria-live="polite"
        id="chakra-toast-manager-bottom-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-accordion css-1wbdjbz"
    >
      <div
        class="css-0"
      >
        <p
          class="css-0"
        >
          Accordion Test
        </p>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-svjswr"
      />
      <div
        class="css-0"
      >
        <div
          class="chakra-accordion__item css-17mg6aq"
        >
          <h2>
            <button
              aria-controls="accordion-panel-:r0:"
              aria-expanded="false"
              class="chakra-accordion__button css-uttm9k"
              data-index="0"
              id="accordion-button-:r0:"
              type="button"
            >
              <div
                class="css-0"
              >
                Accordion Item Title
              </div>
              <svg
                aria-hidden="true"
                class="chakra-icon chakra-accordion__icon css-186l2rg"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </h2>
          <div
            class="chakra-collapse"
            style="overflow: hidden; display: none; opacity: 0; height: 0px;"
          >
            <div
              aria-labelledby="accordion-button-:r0:"
              class="chakra-accordion__panel css-mxa4qn"
              id="accordion-panel-:r0:"
              role="region"
            >
              <p
                class="chakra-text css-0"
              >
                Example children
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <span
      hidden=""
      id="__chakra_env"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<AccordionItem /> should render custom title: customTitle 1`] = `
<DocumentFragment>
  <div
    class="chakra-accordion css-1wbdjbz"
  >
    <div
      class="css-0"
    >
      <p
        class="css-0"
      >
        Accordion Test
      </p>
    </div>
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-svjswr"
    />
    <div
      class="css-0"
    >
      <div
        class="chakra-accordion__item css-17mg6aq"
      >
        <div
          class="css-1ki54i"
        >
          <div
            class="chakra-stack css-lxnwxr"
            data-testid="custom-title"
          >
            <div
              class="chakra-stack css-84zodg"
            >
              <button
                aria-controls="accordion-panel-:r2:"
                aria-expanded="false"
                class="chakra-accordion__button css-uttm9k"
                data-index="0"
                data-testid="custom-title-button"
                id="accordion-button-:r2:"
                type="button"
              >
                <h2>
                  Admin
                </h2>
                <svg
                  aria-hidden="true"
                  class="chakra-icon chakra-accordion__icon css-186l2rg"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    fill="currentColor"
                  />
                </svg>
              </button>
              <div
                class="chakra-stack css-u0xykq"
              >
                <button
                  class="css-xtw3c9"
                >
                  <span
                    class="css-14ftdz1"
                  >
                    Complement
                  </span>
                </button>
              </div>
            </div>
            <div
              class="chakra-stack css-84zodg"
            >
              <button
                class="chakra-button css-de9zzv"
                type="button"
              >
                Click
              </button>
            </div>
          </div>
        </div>
        <div
          class="chakra-collapse"
          style="overflow: hidden; display: none; opacity: 0; height: 0px;"
        >
          <div
            aria-labelledby="accordion-button-:r2:"
            class="chakra-accordion__panel css-mxa4qn"
            id="accordion-panel-:r2:"
            role="region"
          >
            <p
              class="chakra-text css-0"
            >
              Example children
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
